src/manager/checkpoint/manager/topdown.rs

```diff
@@ -22,47 +93,135 @@ impl CheckpointManager for TopDownCheckpointManager {
 
     fn checkpoint_period(&self) -> ChainEpoch {
-        todo!()
+        self.checkpoint_period
     }
 
     async fn last_executed_epoch(&self) -> anyhow::Result<ChainEpoch> {
-        todo!()
+        let child_gw_state = self.sync_subnet_gateway_state().await?;
+        Ok(child_gw_state
+            .top_down_checkpoint_voting
+            .last_voting_executed)
     }
 
     async fn current_epoch(&self) -> anyhow::Result<ChainEpoch> {
-        todo!()
+        let parent_head = self.parent_client.chain_head().await?;
+        Ok(ChainEpoch::try_from(parent_head.height)?)
     }
 
     async fn submit_checkpoint(
         &self,
-        _epoch: ChainEpoch,
-        _previous_epoch: ChainEpoch,
-        _validator: &Address,
+        epoch: ChainEpoch,
+        validator: &Address,
     ) -> anyhow::Result<()> {
-        todo!()
+        let nonce = self
+            .child_client
+            .ipc_read_gateway_state(self.child_tipset().await?)
+            .await?
+            .applied_topdown_nonce;
+
+        let submission_tip_set = self.submission_tipset(epoch).await?;
+        let top_down_msgs = self
+            .parent_client
+            .ipc_get_topdown_msgs(
+                &self.child_subnet.id,
+                &self.parent.gateway_addr,
+                submission_tip_set,
+                nonce,
+            )
+            .await?;
+
+        log::debug!(
+            "nonce: {:} for submission tip set: {:} at epoch {:} of manager: {:}, size of top down messages: {:}",
+            nonce, submission_tip_set, epoch, self, top_down_msgs.len()
+        );
+
+        // we submit the topdown messages to the CHILD subnet.
+        let topdown_checkpoint = TopDownCheckpoint {
+            epoch,
+            top_down_msgs,
+        };
+        let message = MpoolPushMessage::new(
+            self.parent.gateway_addr,
+            *validator,
+            ipc_gateway::Method::SubmitTopDownCheckpoint as MethodNum,
+            cbor::serialize(&topdown_checkpoint, "topdown_checkpoint")?.to_vec(),
+        );
+        let mem_push_response = self
+            .child_client
+            .mpool_push_message(message)
+            .await
+            .map_err(|e| {
+                log::error!("error submitting checkpoint at epoch {epoch:} for manager: {self:}");
+                e
+            })?;
+
+        // wait for the checkpoint to be committed before moving on.
+        let message_cid = mem_push_response.cid()?;
+        log::debug!(
+            "checkpoint at epoch {:} for manager: {:} published with cid: {:?}, wait for execution",
+            epoch,
+            self,
+            message_cid,
+        );
+        self.child_client.state_wait_msg(message_cid).await?;
+        log::debug!(
+            "checkpoint at epoch {:} for manager: {:} published with cid: {:?}, executed",
+            epoch,
+            self,
+            message_cid,
+        );
+
+        Ok(())
     }
 
     async fn should_submit_in_epoch(
```

Comment:
I don't know if this also exists for the bottom-up checkpoint. If this is not the case we should also check it here. Again leaving the comment before reviewing the corresponding PR so I don't forget :)